CRM/Civioffice/MimeType.php

```diff
@@ -1,74 +1,1879 @@
+ * | Copyright (C) 2021 SYSTOPIA                            |
+ * | Author: J. Franz (franz@systopia.de)                   |
+ * +--------------------------------------------------------+
+ * | This program is released as free software under the    |
+ * | Affero GPL license. You can redistribute it and/or     |
+ * | modify it under the terms of this license which you    |
+ * | can read by viewing the included agpl.txt or online    |
+ * | at www.gnu.org/licenses/agpl.html. Removal of this     |
+ * | copyright header is strictly prohibited without        |
+ * | written permission from the original author(s).        |
+ * +-------------------------------------------------------
+ */
+final class CRM_Civioffice_MimeType {
 
-    public const PDF = 'application/pdf';
+  public const ODT = 'application/vnd.oasis.opendocument.text';
 
-    public const ZIP = 'application/zip';
+  public const DOCX = 'application/vnd.openxmlformats-officedocument.wordprocessingml.document';
 
-    /**
-     * Map te MIME type to the file ending without a pre dot
-     *
-     * @param $mime_type
-     *
-     * @return string
-     *   file ending like docx or pdf
-     * @throws \InvalidArgumentException
-     */
-    public static function mapMimeTypeToFileExtension($mime_type): string
-    {
-        $mapping = self::mimeTypeToFileExtensionMap();
-        if (isset($mapping[$mime_type])) {
-            return $mapping[$mime_type];
-        } else {
-            throw new InvalidArgumentException(sprintf('Missing implementation for MIME type "%s"', $mime_type));
-        }
-    }
+  public const PDF = 'application/pdf';
 
-    /**
-     * Get a list of MIME type to file extension
-     *
-     * @return array
-     */
-    public static function mimeTypeToFileExtensionMap(): array
-    {
-        return [
-            self::PDF  => 'pdf',
-            self::DOCX => 'docx',
-            self::ZIP  => 'zip',
-        ];
-    }
+  public const ZIP = 'application/zip';
+
+  public const PNG = 'image/png';
+
+  public const RTF = 'text/rtf';
 
-    /**
-     * Checks if the file ending/extension matches with the given fully qualified MIME type
-     *
-     * MIME type checks could be handled differently in the future: https://github.com/systopia/de.systopia.civioffice/issues/2
-     *
-     * @param $file_name
-     * @param $mime_type
-     *
-     * @return bool Returns true if given MIME type is equal to ending/extension
-     * @throws \Exception
-     */
-    public static function hasSpecificFileNameExtension($file_name, $mime_type): bool
-    {
-        $extension = self::mapMimeTypeToFileExtension($mime_type);
+  /**
+   * A map of MIME types and their default extensions.
+   *
+   * Source: Symfony Mime component.
+   *
+   * @see https://github.com/symfony/symfony/blob/cadcc578af9ad9fe7a8dc87aa545083f23950e12/src/Symfony/Component/Mime/MimeTypes.php#L142
+   *
+   * phpcs:disable Generic.Files.LineLength.TooLong
+   */
+  private const MAP = [
```

Comment:
How about using a dependency like https://github.com/ralouphie/mimey or a parser for [this file](https://svn.apache.org/repos/asf/httpd/httpd/trunk/docs/conf/mime.types)?